### ISSUE TITLE
Easy use of rule editor: Support 'key press' autofill via keyboard button press

### ISSUE THREAD

**Issue by MattHag** [pwr-Solaar/Solaar#2267]
**Information**

- Solaar version: 1.1.11rc1-3-ge1ac8b5

**Is your feature request related to a problem? Please describe.**
Configuring key presses in the rule editor is not beginner-friendly and intuitive, as every key has to be selected manually from a list. Names like Super_L for the left Windows/CMD key has to be known in advance. 

A huge improvement would be the detection of key presses and auto-fill of this key combination - at least for the most common keys a-z, Super, Alt, Shift.

**Comment by pfps** [comment]
Yes this would be useful.

**Comment by pfps** [comment]
The question is how to do this while still allowing input of other keys.  One of the reasons to have this action is to allow simulation of keys that are not on the (current) keyboard.

**Comment by MattHag** [comment]
Yeah sure, that makes sense. Thinking quickly about a button could enable the detection and pre fill the existing UI, then you can add and remove whatever you want.

**Comment by pfps** [comment]
A problem with this is that the Solaar GUI does not see the raw keypresses.   Instead what it sees is the result of processing by the Linux input system and by the windowing system and maybe even by other system programs.

**Comment by MattHag** [comment]
But somewhere this configuration ant the input need to match and be detectable, so there's certainly a solution for this.

**Comment by pfps** [comment]
The mapping is the result of a lot of code.  Reversing it is problematic.   Most programs need to know nothing about it.   Some programs get to see more.  But as far as I know there is no program that reliably can simulate keypresses that result in a particular character being seen by regular programs.
